src/ensembl/src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewActions.ts

```diff
@@ -0,0 +1,127 @@
+import { createAction } from 'typesafe-actions';
+import { ActionCreator, Action } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+
+import {
+  getEntityViewerActiveGenomeId,
+  getEntityViewerActiveEnsObjectId
+} from 'src/content/app/entity-viewer/state/general/entityViewerGeneralSelectors';
+import {
+  EntityViewerGeneViewUIStateState,
+  GeneViewTabName,
+  GeneFunctionTabName,
+  GeneRelationshipsTabName,
+  EntityViewerGeneFunctionState,
+  EntityViewerGeneRelationshipsState
+} from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewState';
+
+import { RootState } from 'src/store';
+
+export const updateActiveGeneViewObjectState = createAction(
+  'entity-viewer/update-active-gene-view-object-state'
+)<{
+  activeGenomeId: string;
+  activeObjectId: string;
+  fragment: Partial<EntityViewerGeneViewUIStateState>;
+}>();
+
+export const setActiveGeneTab: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = (selectedTabName: GeneViewTabName) => (
+  dispatch,
+  getState: () => RootState
+) => {
+  const activeGenomeId = getEntityViewerActiveGenomeId(getState());
+  const activeObjectId = getEntityViewerActiveEnsObjectId(getState());
+
+  if (!activeGenomeId || !activeObjectId) {
+    return;
+  }
+
+  dispatch(
+    updateActiveGeneViewObjectState({
+      activeGenomeId,
+      activeObjectId,
+      fragment: {
+        selectedGeneTabName: selectedTabName
+      }
+    })
+  );
+};
+
+export const updateActiveGeneViewObjectGeneFunctionState = createAction(
+  'entity-viewer/update-active-gene-view-object-gene-function-state'
+)<{
+  activeGenomeId: string;
+  activeObjectId: string;
+  fragment: Partial<EntityViewerGeneFunctionState>;
+}>();
+
+export const setActiveGeneFunctionTab: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = (selectedTabName: GeneFunctionTabName) => (
+  dispatch,
+  getState: () => RootState
+) => {
+  const activeGenomeId = getEntityViewerActiveGenomeId(getState());
+  const activeObjectId = getEntityViewerActiveEnsObjectId(getState());
+
+  if (!activeGenomeId || !activeObjectId) {
+    return;
+  }
+
+  dispatch(
+    updateActiveGeneViewObjectState({
+      activeGenomeId,
+      activeObjectId,
+      fragment: {
+        geneFunction: {
+          selectedTabName
+        }
+      }
+    })
+  );
+};
+
+export const updateActiveGeneViewObjectGeneRelationshipsState = createAction(
```

Comment:
Isn't this a very long variable name?

Reply:
Yes, it is. I am not sure how to shorten it. Any suggestions?

Reply:
Oh gosh, yes.
`updateGeneRelationshipsUI`?

Reply:
Just realised that this actionCreator is no longer needed. I've removed it. Sorry!